lib/vm.py

```diff
@@ -178,31 +177,45 @@ def exists(self):
     def exists_on_previous_pool(self):
         return self.previous_host.pool_has_vm(self.uuid)
 
-    def migrate(self, target_host, sr=None):
-        # workaround XO bug where sometimes it loses connection without knowing it
-        self.host.pool.master.xo_server_reconnect()
-        if target_host.pool != self.host.pool:
-            target_host.pool.master.xo_server_reconnect()
-
-        # Sometimes we migrate VMs right after creating them
-        # In that case we need to ensure that XO knows about the new VM
-        # Else we risk getting a "no such VM" error
-        # Thus, let's first wait for XO to know about the VM
-        wait_for(lambda: xo_object_exists(self.uuid), "Wait for XO to know about VM %s" % self.uuid)
-
+    def migrate(self, target_host, sr=None, network=None):
         msg = "Migrate VM to host %s" % target_host
         params = {
-            'vm': self.uuid,
-            'targetHost': target_host.uuid
+            'uuid': self.uuid,
+            'host-uuid': target_host.uuid,
+            'live': self.is_running()
         }
+        cross_pool = self.host.pool.uuid != target_host.pool.uuid
         if sr is not None:
             msg += " (SR: %s)" % sr.uuid
-            mapping = {}
-            for vdi_uuid in self.vdi_uuids():
-                mapping[vdi_uuid] = sr.uuid
-            params['mapVdisSrs'] = 'json:' + json.dumps(mapping)
+        if network is not None:
+            msg += " (Network: %s)" % network
         logging.info(msg)
-        xo_cli('vm.migrate', params)
+
+        storage_motion = cross_pool or sr is not None or network is not None
```

Comment:
I'd rename `storage_motion` to `migrate_send`. Storage Motion as a marketing term means the VDI will be moved. This is not true if the provenance SR and destination SR are the same (such as a shared SR).

Reply:
I disagree, here's the log output from the migrate call:
```
Jan 23 12:14:17 DEBUG [172.16.210.26] xe vm-migrate uuid=e099775a-1485-91ae-4d85-fe9bcb3094c2 host-uuid=f8b9daf0-67b3-427a-966b-3c7754216dac live=true remote-master=remote_ip remote-username=root remote-password=dummy_pwd vdi:3db4af3d-9c30-4408-b50e-2a1042c33e32=6a6b3a26-2129-16f4-7412-e780912fa232 vif:ee23447b-5b83-ba34-6b02-b13c4136abc5=89117c9a-4c8e-25cc-9ccc-9afcbd4d5065
Performing a Storage XenMotion migration. Your VM's VDIs will be migrated with the VM.
```

In this case (even with same shared SR).